ethdb/privateapi/ethbackend.go

```diff
@@ -270,6 +285,56 @@ func (s *EthBackendServer) EngineGetPayloadV1(ctx context.Context, req *remote.E
 	return nil, fmt.Errorf("unknown payload")
 }
 
+// EngineForkChoiceUpdatedV1, either states new block head or request the assembling of a new bloc
+func (s *EthBackendServer) EngineForkChoiceUpdatedV1(ctx context.Context, req *remote.EngineForkChoiceUpdatedRequest) (*remote.EngineForkChoiceUpdatedReply, error) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	if s.config.TerminalTotalDifficulty == nil {
+		return nil, fmt.Errorf("not a proof-of-stake chain")
+	}
+	// Check if parent equate to the head
+	parent := gointerfaces.ConvertH256ToHash(req.Forkchoice.HeadBlockHash)
+	tx, err := s.db.BeginRo(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	headHeader, err := rawdb.ReadHeaderByHash(tx, parent)
+	if err != nil {
+		return nil, err
+	}
+
+	if atomic.LoadUint32(s.waitingForBeaconChain) == 0 || headHeader == nil {
+		return &remote.EngineForkChoiceUpdatedReply{
+			Status: string(Syncing),
+		}, nil
+	}
+
+	// Hash is incorrect because mining archittecture has yet to be implemented
+	s.pendingPayloads[s.nextPayloadId] = types2.ExecutionPayload{
+		ParentHash:    req.Forkchoice.HeadBlockHash,
+		Coinbase:      req.Prepare.FeeRecipient,
+		Timestamp:     req.Prepare.Timestamp,
+		Random:        req.Prepare.Random,
+		StateRoot:     gointerfaces.ConvertHashToH256(headHeader.Root),
+		ReceiptRoot:   gointerfaces.ConvertHashToH256(types.EmptyRootHash),
+		LogsBloom:     &types2.H2048{},
+		GasLimit:      headHeader.GasLimit,
+		GasUsed:       0,
+		BlockNumber:   headHeader.Number.Uint64() + 1,
+		ExtraData:     headHeader.Extra,
+		BaseFeePerGas: &types2.H256{},
+		BlockHash:     gointerfaces.ConvertHashToH256(headHeader.Hash()),
+		Transactions:  [][]byte{},
+	}
+	// successfully assembled the payload and assinged the correct id
+	defer func() { s.nextPayloadId++ }()
+	return &remote.EngineForkChoiceUpdatedReply{
+		Status:    "SUCCESS",
+		PayloadId: s.nextPayloadId,
```

Comment:
Suggestion: rename `nextPayloadId` to `payloadId` because it's used as the ID of the current payload, not the next one.

Reply:
done